gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -322,6 +326,18 @@ public AddSpecResponse onAddSpec(Spec addedSpec) {
       return new AddSpecResponse<>(response);
     }
 
+    // Check quota limits against run immediately flows or adhoc flows before saving the schedule
+    if (!jobConfig.containsKey(ConfigurationKeys.JOB_SCHEDULE_KEY) || PropertiesUtils.getPropAsBoolean(jobConfig, ConfigurationKeys.FLOW_RUN_IMMEDIATELY, "false")) {
+      try {
+        if (quotaManager.isPresent()) {
+          quotaManager.get().checkQuota(dag.getNodes().get(0), false);
```

Comment:
Will this cause us to double the quota for this job since we check quota again when submit the job?

Reply:
It wouldn't since in the QuotaManager there's a check against duplicate dagIds to not double increment if the dag was already added. I can add a comment here to reflect this behavior. It seems a bit hacky but I couldn't think of another way of handling this scenario since the dagmanager asynchronous design doesn't make it easy to validate that a flow is running on the specExecutor before returning a http status code.

Reply:
yeah, add one comment will help understanding for sure, let's do that

Reply:
Yes it ensures that dags are only counted once for each dagId (so every unique flowexecution)